changelog/changelog.py

```diff
@@ -278,21 +357,41 @@ def main():
     ]
 
     changelog_pr = set()
-    generator, labels = get_custom_options(repo)
+    generator, labels, header_labels = get_custom_options(repo)
     if args.labels is not None:
         labels = args.labels
-
-    for label in labels:
-        pull_requests = api.search(milestone['title'], label)
-        items = pull_requests['items']
-        if items:
-            if label:
-                lines.extend([
-                    "\n",
-                    f"**{label.upper()}**\n",
-                ])
-            changelog_pr = changelog_pr.union(pr['html_url'] for pr in items)
-            lines.extend(generator(items))
+    if args.header_labels is not None:
+        header_labels = args.header_labels
+    if header_labels is None:
+        header_labels = []
+
+    if header_labels:
+        for header_label in header_labels:
+            pull_requests = api.search(milestone['title'], header_label)
+            items = pull_requests['items']
+            if items:
+                # If a single header is given as an input, don't add H3
+                if header_label and len(header_labels) >= 2:
+                    lines.extend([
+                        "\n",
+                        f"### {header_label.upper()}\n",
+                    ])
+
+                some_lines, some_changelog_pr = default_header_changelog_generator(items, labels, generator)
+                lines.extend(some_lines)
+                changelog_pr = changelog_pr.union(some_changelog_pr)
```

Comment:
Just like we can `.extend()` a `list`, we can `.update()` a `set` to add items to the existing object.
```suggestion
                changelog_pr.update(some_changelog_pr)
```